### DISCUSSION TITLE
Implement a feature for fallback from one protocol to another

### DISCUSSION THREAD

**Discussion by kamyar1979** [icerpc/icerpc-csharp#3949] [Ideas]
Since QUIC works on UDP, we may have problems using some Internet Service Providers. If we can use a feature which allows us to provide a list of protocols by priority, it would be great! For example: Quic, Slic, Tcp...

**Comment by bernardnormier** [comment]
We can actually already do this with IceRPC - at the application level.

On the server-side, you just need to create two servers that share the same dispatch pipeline. One server listens on QUIC while the other server listens on TCP (+ Slic to get the icerpc protocol) - it's straightforward.

On the client-side, one simple solution is to create a ClientConnection that attempts a QUIC connection, and if this connection attempts fails, we fall back to TCP. 

Since a picture is worth a thousand words, I added an example that illustrates the above:
https://github.com/icerpc/icerpc-csharp/tree/0.3.x/examples/slice/TcpFallback (Slice version)
https://github.com/icerpc/icerpc-csharp/tree/0.3.x/examples/protobuf/TcpFallback (Protobuf version)

This fallback ability doesn't require much code. What do you think?
